typescript/sdk/src/hook/EvmHookReader.ts

```diff
@@ -125,13 +121,13 @@ export class EvmHookReader implements HookReader {
           );
       }
     } catch (e) {
-      this.logger.debug(
-        `Failed to derive ${onchainHookType} hook (${address}): ${e}`,
+      this.logger.error(
+        `Failed to derive ${onchainHookType} hook (${address}):`,
+        e,
       );
-    } finally {
       this.setSmartProviderLogLevel(getLogLevel()); // returns to original level defined by rootLogger
+      throw e;
```

Comment:
This is fine but this pattern of log and rethrow is generally what the error `cause` field is for: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
I think the cause could be useful down the line, if we need to do some error based actions